File: 프로그래머스/Lv.3/12904. 가장 긴 팰린드롬/가장 긴 팰린드롬.java
class Solution
{
    public int solution(String s) {
        int answer = 1;
        for(int i = 0; i < s.length() - 1; i++) {
            for (int j = i+1; j < s.length(); j++) {
               if(isPalindrome(s, i, j) && answer < j - i + 1) {
                   answer = j - i + 1;
               }
            }
        }
        return answer;
    }
    
    public boolean isPalindrome(String s, int left, int right) {
        while (left < right) {
            if (s.charAt(left) != s.charAt(right)) {
                return false;
            }
            left++;
            right--;
        }
        
        return true;
    }
}

File: 프로그래머스/Lv.3/12904. 가장 긴 팰린드롬/가장 긴 팰린드롬.py
def solution(s):
    answer = 1
    for i in range(len(s) - 1):
        for j in range(i + 1, len(s)):
            target = s[i:j+1]
            
            if target == target[::-1]:
                answer = max(answer, j - i + 1)
                
    return answer